### PR TITLE
Moves all ranged stuns and the improvised stun prod to a stamina based system

### PR DESCRIPTION
**Pull request by GuyonBroadway**
:cl: Guyonbroadway
balance: Taser bolts now deal 80 stamina damage and force people to drop items.
balance: Improvised stun prods deal 80 stamina damage that ignores armour and force people to drop items.
balance: The detectives revolver now deals 65 stamina damage, 15 brute and forces people to drop items.
balance: Taser and disabler beams have had their shot cost adjusted, with a standard taser you have 8 taser bolts or 12 disabler beams. Enough to drop four nerds if your aim is on point. 
balance: Syndiacte Sniper Rifles still have a stun, but it has been reduced to one tick (not that it matters when you are so slow the followup shot is both easy to land and will put you in crit anyway) 
balance: Bulldog shotguns taser rounds also match the tasers change, but they count as bullets so are better vs certain types of defences. 
balance: Yes I took into account the R&D guns. 
balance: Yes the mechs too
experiment: Disabler beams now travel faster so you can peg fleeing targets more easily (You can still just about outrun them on ephedrine though)

/:cl:

[why]:  Hard ranged stuns are very powerful and basically dictate the course of the game.

A stamina based system is less "one stun and done" and might provoke more firefights or more creative methods of combat besides "grab taser - go kill" The makeshift baton was also better than its sec counterpart and widely available. 

The sec baton will remain the hard stunner that it is as it's the final means for sec to detain a perp.

There are other changes I feel could benefit this new system but I want to see how the player-base handles it first, it might just be a bad idea altogether and needs scrapping.

### CODE REVIEW

**Comment by GuyonBroadway** [comment]
I missed out on the mechs file, hold off before merging.

**Comment by tortellinitony** [comment]
You're still deleting the submodule. Fix that.

File: hippiestation.dme [hunk 1]
```diff
@@ -1466,6 +1466,7 @@
 #include "code\modules\mob\mob_movement.dm"
 #include "code\modules\mob\mob_transformation_simple.dm"
 #include "code\modules\mob\say.dm"
+#include "code\modules\mob\say_readme.dm"
```
Comment (tortellinitony) [inline hippiestation.dme]: Why is this enabled?

File: hippiestation.dme [hunk 2]
```diff
@@ -22,6 +22,7 @@
 #include "code\__DATASTRUCTURES\linked_lists.dm"
 #include "code\__DATASTRUCTURES\priority_queue.dm"
 #include "code\__DATASTRUCTURES\stacks.dm"
+#include "code\__DEFINES\_readme.dm"
```
Comment (tortellinitony) [inline hippiestation.dme]: Why is this enabled?

File: hippiestation.dme [hunk 3]
```diff
@@ -1244,7 +1245,6 @@
 #include "code\modules\events\spontaneous_appendicitis.dm"
 #include "code\modules\events\vent_clog.dm"
 #include "code\modules\events\wormholes.dm"
-#include "code\modules\events\holiday\halloween.dm"
```
Comment (tortellinitony) [inline hippiestation.dme]: Why is this disabled?

File: .gitmodules [hunk 4]
```diff
@@ -2,3 +2,6 @@
 	path = tools/tgstation-server
 	url = https://github.com/tgstation/tgstation-server
 	branch = master
+[submodule "tools/tgstation-server-tools"]
```
Comment (tortellinitony) [inline .gitmodules]: Delete this change.

**Review by tortellinitony — CHANGES_REQUESTED** [review]
I also don't get why you aren't just changing the values for the actual projectiles.

**Comment by GuyonBroadway** [comment]
And just so you know, yes it's been tested, yes it works, no there are no runtimes.

**Comment by j-mie** [comment]
I want a test merge of this in a couple of days.

**Comment by PopNotes** [comment]
I agree with absolutely everything here. Guy is our savior.

**Comment by Uqualm** [comment]
For posterity's sake, let it be known I think this is a terrible idea. I also think that GuyonBroadway is so in love with his idea that he won't see any of the flaws in it and even if it's pure shit (like anti-stuns) it'll stay in the game. Yeah I know nobody cares what I think.

**Comment by lcass** [comment]
welp hes got conflicts.

**Comment by tortellinitony** [comment]
Also if this is about ranged stuns, then **everything involving the stunbatons and stunprods should also be in its own PR if you want this to be mergeable**

**Comment by ghost** [comment]
I'm for trialling this in a bit later when our population is a bit higher and I'm on. I would want to see it on a bigger player pool and see how people feel about it for a few rounds and get feedback.

While I do support these changes on the whole, they're very drastic and we should be able to revert for any reason during the testing period based on feedback and make improvements.

**Comment by KayeArray** [comment]
MUH SNIPER NERF

**Comment by lcass** [comment]
I think you should reduce the damage of the detectives revolver to 10 brute as it will take 2 shots minimum to down someone thats a third of their health gone, seems a bit too easy to crit with.

**Comment by PopNotes** [comment]
god, please merge this already

**Comment by ghost** [comment]
I'm for trialing this on a day or so I can be on for a while.

File: hippiestation/code/game/objects/items/weapons/stunbaton.dm [hunk 5]
```diff
@@ -0,0 +1,40 @@
+
+/obj/item/weapon/melee/baton
+	var/stamforce = 80
+
```
Comment (KayeArray) [inline hippiestation/code/game/objects/items/weapons/stunbaton.dm:4]: use a define

File: hippiestation/code/modules/projectiles/projectile/bullets.dm [hunk 6]
```diff
@@ -0,0 +1,14 @@
+/obj/item/projectile/bullet/weakbullet2
+	weaken = 0
+	stun = 0.1
+	stamina = 65 //Plus the 15 base damage means two shots will down a perp
+
+
+/obj/item/projectile/bullet/stunshot
```
Comment (KayeArray) [inline hippiestation/code/modules/projectiles/projectile/bullets.dm:7]: is nerfing this necessary?

File: hippiestation/code/modules/projectiles/projectile/energy.dm [hunk 7]
```diff
@@ -0,0 +1,10 @@
+/obj/item/projectile/energy/electrode
+	stun = 0.1
+	weaken = 0
+	stamina = 80
+	range = 8
```
Comment (KayeArray) [inline hippiestation/code/modules/projectiles/projectile/energy.dm]: you buffed the range

Comment (GuyonBroadway) [reply hippiestation/code/modules/projectiles/projectile/energy.dm]: And just by one tile too... I'll remove that but they might need a range buff later down the line

File: hippiestation/code/game/objects/items/weapons/stunbaton.dm (hunk 5 above)
Comment (GuyonBroadway) [reply hippiestation/code/game/objects/items/weapons/stunbaton.dm:4]: This is exactly how stunpower works for regular batons, I need to do it like this lest I need to balance different things specifically.

File: hippiestation/code/modules/projectiles/projectile/bullets.dm (hunk 6 above)
Comment (GuyonBroadway) [reply hippiestation/code/modules/projectiles/projectile/bullets.dm:7]: Yes, if we are changing ONE ranged stun we need to change them all.

**Pull request merged**